### PR TITLE
[Enhance] Enhance `IndoorPatchPointSample` transform

### LINKED ISSUE #608: Enhance `IndoorPatchPointSample`
- Add size threshold control
- Support flexible break condition
- Remove useless param `sample_rate`

### PR DESCRIPTION
**Pull request by Wuziyi616**
Fix #608. Enhance `IndoorPatchPointSample` transform by:
- Remove useless param `sample_rate` (this is actually used in sliding inference of segmentors)
- Support more flexible point selection criterion by changeable size range
- Support more flexible point uniqueness judging standard

The later two are required in PAConv training. Stay tuned with PAConv model.

### CODE REVIEW

File: configs/pointnet2/pointnet2_msg_xyz-only_16x2_cosine_250e_scannet_seg-3d-20class.py [hunk 1]
```diff
@@ -116,8 +117,7 @@
         classes=class_names,
         test_mode=False,
         ignore_index=len(class_names),
-        scene_idxs=data_root + 'seg_info/train_resampled_scene_idxs.npy',
-        label_weight=data_root + 'seg_info/train_label_weight.npy'),
```
Comment (Wuziyi616) [inline configs/pointnet2/pointnet2_msg_xyz-only_16x2_cosine_250e_scannet_seg-3d-20class.py:120]: I forgot to delete `label_weight` in PR #607. Remove it here.

File: mmdet3d/datasets/pipelines/transforms_3d.py [hunk 2]
```diff
@@ -905,30 +905,42 @@ class IndoorPatchPointSample(object):
...
-                 sample_rate=1.0,
                  ignore_index=None,
                  use_normalized_coord=False,
-                 num_try=10):
+                 num_try=10,
+                 enlarge_size=0.2,
+                 min_unique_num=None):
         self.num_points = num_points
         self.block_size = block_size
-        self.sample_rate = sample_rate
```
Comment (ZwwWayne) [inline mmdet3d/datasets/pipelines/transforms_3d.py:928]: We should still accept sample_rate and add a deprecated warning first to inform users to modify their code.

File: docs/compatibility.md [hunk 3]
```diff
@@ -2,6 +2,12 @@
 
 This document provides detailed descriptions of the BC-breaking changes in MMDetection3D.
 
+## MMDetection3D 0.15.0
+
+### Enhance `IndoorPatchPointSample` transform
+
+We enhance the pipeline function `IndoorPatchPointSample` by adding more choices for patch selection and removing a useless parameter `sample_rate`. Please modify the code as well as the config files accordingly if you use this transform.
```
Comment (ZwwWayne) [inline docs/compatibility.md:9]: We plan to remove the unused parameter `sample_rate` in the future.

File: mmdet3d/datasets/pipelines/transforms_3d.py (hunk 2 above)
Comment (Wuziyi616) [reply mmdet3d/datasets/pipelines/transforms_3d.py:928]: Keep sample_rate and add warning.

File: docs/compatibility.md (hunk 3 above)
Comment (Wuziyi616) [reply docs/compatibility.md]: Modified.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**